src/main/java/com/google/roomies/servlets/ListingsServlet.java

```diff
@@ -80,4 +84,60 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) {
         response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
     }
   }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response)
+      throws IOException {
+    database = DatabaseFactory.getDatabase();
+    try {
+      List<Listing> listings = getAllListingsFromCollection();
+
+      response.setContentType("application/json");
+      response.getWriter().println(convertToJsonUsingGson(listings));
+    } catch (Exception e) {
```

Comment:
What exceptions do you expect here? This catch block can be more narrowly scoped.